contracts/implementations/lazyMintNative/RMRKEquippableLazyMintNative.sol

```diff
@@ -0,0 +1,126 @@
+            data.maxSupply,
+            data.royaltyRecipient,
+            data.royaltyPercentageBps
+        )
+    {
+        _pricePerMint = data.pricePerMint;
+    }
+
+    /**
+     * @notice Used to mint the desired number of tokens to the specified address.
+     * @dev The `data` value of the `_safeMint` method is set to an empty value.
+     * @dev Can only be called while the open sale is open.
+     * @param to Address to which to mint the token
+     * @param numToMint Number of tokens to mint
+     * @return The ID of the first token to be minted in the current minting cycle
+     */
+    function mint(
+        address to,
+        uint256 numToMint
+    ) public payable virtual returns (uint256) {
+        (uint256 nextToken, uint256 totalSupplyOffset) = _prepareMint(
+            numToMint
+        );
+        _chargeMints(numToMint);
+
+        for (uint256 i = nextToken; i < totalSupplyOffset; ) {
+            _safeMint(to, i, "");
+            unchecked {
+                ++i;
+            }
+        }
+
+        return nextToken;
+    }
+
+    /**
+     * @notice Used to mint a desired number of child tokens to a given parent token.
+     * @dev The `data` value of the `_safeMint` method is set to an empty value.
+     * @dev Can only be called while the open sale is open.
+     * @param to Address of the collection smart contract of the token into which to mint the child token
+     * @param numToMint Number of tokens to mint
+     * @param destinationId ID of the token into which to mint the new child token
+     * @return The ID of the first token to be minted in the current minting cycle
+     */
+    function nestMint(
+        address to,
+        uint256 numToMint,
+        uint256 destinationId
+    ) public payable virtual returns (uint256) {
+        (uint256 nextToken, uint256 totalSupplyOffset) = _prepareMint(
+            numToMint
+        );
+        _chargeMints(numToMint);
+
+        for (uint256 i = nextToken; i < totalSupplyOffset; ) {
+            _nestMint(to, i, destinationId, "");
+            unchecked {
+                ++i;
+            }
+        }
+
+        return nextToken;
+    }
+
+    function _chargeMints(uint256 numToMint) internal {
+        uint256 price = numToMint * _pricePerMint;
+        if (price != msg.value) revert RMRKWrongValueSent();
+    }
+
+    /**
+     * @notice Used to retrieve the price per mint.
+     * @return The price per mint of a single token expressed in the lowest denomination of a native currency
+     */
+    function pricePerMint() public view returns (uint256) {
+        return _pricePerMint;
+    }
+
+    /**
+     * @notice Used to withdraw the minting proceedings to a specified address.
+     * @dev This function can only be called by the owner.
+     * @param to Address to receive the given amount of minting proceedings
+     * @param amount The amount to withdraw
+     */
+    function withdrawRaised(address to, uint256 amount) external onlyOwner {
```

Comment:
Same comment on withdrawl. I do not see any need.

Reply:
I guess draining this would require some social engineering, but it could trigger a malicious fallback function in the destination. Well.. the same argument from my side as well. If we validated through tx.origin, then a reentrancy guard would be necessary; here, it is just a remote possibility that someone might do some annoying things 😅 

So you could do the pig to a slaughter attack here, but if you trust someone enough to do this, you might as well give them your private key anyway 😂

Reply:
If there's some heavy social engineering they would have to target all multisig signers and if they trick them into sign this, well, no reentrancy will prevent it :grimacing: